rpm/generic/zfs.spec.in

```diff
@@ -34,31 +34,14 @@
 %endif
 
 %bcond_with    debug
-%bcond_with    systemd
+%bcond_without systemd
 
-# Generic enable switch for systemd
-%if %{with systemd}
+# Enable systemd when _unitdir is defined or with generic enable switch.
+%if %{with systemd} && %{defined _unitdir}
 %define _systemd 1
```

Comment:
Since you're changing systemd related flags in the spec file, it may be worth doing some clean up while you're here. You can remove the `%define _systemd 1` and replace it with `%define with_systemd 1`. Then all the checks to determine if systemd is enabled will become `%if %{with systemd}`. Thoughts?

See https://review.whamcloud.com/#/c/27215/5/lustre.spec.in for an example.